### PR TITLE
refactor: split SQLBindParameter C-to-VARCHAR tests into separate files

### PR DESCRIPTION
**Pull request by sfc-gh-makowalski**
## Summary

Restructure `SQLBindParameter` C-to-VARCHAR e2e tests to follow the same file-per-conversion pattern used by the sql-to-c tests (e.g., `real_conversion_to_c_char.cpp`, `fixed_conversion_to_c_integer.cpp`).

Context: [PR #748 review comment from @sfc-gh-pbulawa](https://github.com/snowflakedb/universal-driver/pull/748#discussion_r2069371661):

> We should restructure the binding tests to follow a similar conversion for the sql-to-c tests as it will benefit in readability, however that's not a part of this PR

## Changes

- **Split** the monolithic `sql_bind_parameter.cpp` C-to-VARCHAR tests into 7 focused files:
    - `c_integer_to_varchar.cpp` — SQL_C_LONG, SLONG, ULONG, SSHORT, USHORT, SBIGINT, UBIGINT, STINYINT, UTINYINT (18 tests)
    - `c_real_to_varchar.cpp` — SQL_C_DOUBLE, SQL_C_FLOAT incl. NaN/Inf/-0.0 edge cases (9 tests)
    - `c_bit_to_varchar.cpp` — SQL_C_BIT true/false/overflow (3 tests)
    - `c_char_to_varchar.cpp` — SQL_C_DEFAULT routing + SQL_CHAR/LONGVARCHAR/WCHAR/WVARCHAR/WLONGVARCHAR equivalence (6 tests)
    - `c_temporal_to_varchar.cpp` — SQL_C_TYPE_TIMESTAMP, DATE, TIME structs (4 tests)
    - `c_numeric_to_varchar.cpp` — SQL_NUMERIC_STRUCT with scale/negative scale (3 tests)
    - `c_binary_to_varchar.cpp` — SQL_C_BINARY hex encoding (1 test)
- **Created** corresponding `.feature` files for each new test file
- **Updated** `CMakeLists.txt` to register the new test executables
- **Removed** all C-to-VARCHAR tests and unused includes from `sql_bind_parameter.cpp` (which retains error, lifecycle, and descriptor tests)
- **Extracted** `hex_encode_lowercase` helper in `binary.rs` to eliminate duplicated hex encoding between `varchar.rs` and `binary.rs`
- **Documented** test location rationale for structured C type → VARCHAR pipeline tests in `param_binding.rs`

## Test plan

- [x] All 44 C-to-VARCHAR tests preserved (no test loss)
- [x] `cargo fmt` and `cargo clippy` clean
- [x] All Rust unit tests pass (34 param_binding + 29 binary)
- [x] `.feature` files match 1:1 with C++ test cases

SNOW-3031761

### CODE REVIEW

**Comment by sfc-gh-makowalski** [comment]
* **#768** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/768?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/768?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#749** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/749?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#748** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/748?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: odbc_tests/tests/e2e/query/c_char_to_varchar.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
+#include <string>
+
+#include <catch2/catch_test_macros.hpp>
+
+#include "Connection.hpp"
+#include "get_data.hpp"
+#include "odbc_cast.hpp"
+
+TEST_CASE("should bind SQL_C_CHAR to SQL_VARCHAR.", "[query][bind_parameter][c_char_to_varchar]") {
```
Comment (sfc-gh-ddas) [inline odbc_tests/tests/e2e/query/c_char_to_varchar.cpp:9]: The first 3 test cases do not have the Gherkin style comments that are present everywhere else. We should have Gherkin feature descriptions for consistency.

**Review by sfc-gh-ddas — CHANGES_REQUESTED** [review]

**Review by sfc-gh-ddas — APPROVED** [review]
